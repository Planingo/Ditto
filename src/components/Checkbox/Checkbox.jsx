import React from 'react';
import { Checkbox as AntdCheckbox } from 'antd'


/**
 * 
 * # ðŸš§ _Composant en construction_ ðŸš§
 * 
 * # TODO
 * 
 * ## Description
 * 
 * Documentations andt du composant : https://ant.design/components/checkbox/
 * 
 */
const Checkbox = ({
  props,
  autofocus,
  checked,
  defaultChecked,
  disabled,
  indeterminate,
  onChange,
  children,
}) => {
  return <AntdCheckbox
    aria-label='checkbox'
    autofocus={autofocus}
    checked={checked}
    disabled={disabled}
    defaultChecked={defaultChecked}
    indeterminate={indeterminate}
    onChange={onChange}
    {...props} >{children}</AntdCheckbox>
}

export { Checkbox };