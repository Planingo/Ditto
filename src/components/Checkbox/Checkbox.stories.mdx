import { Meta, ArgsTable } from '@storybook/addon-docs';
import {Checkbox} from './Checkbox';


<Meta
  title="Atomes/Checkbox"
  argTypes={{
    autofocus: {
      name:'autofocus', 
      description:'autofocus', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      },
    },
    checked: {
      description:'checked', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      },
    },
    disabled: {
      description:'disabled', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      },
    },
    defaultChecked: {
      description:'defaultChecked', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      },
    },  
    indeterminate: {
      description:'indeterminate', 
      control: {
        type: 'boolean'
      },
      table: {
        defaultValue: {
          summary: 'false'
        }
      },
    }, 
    onChange: {
      description:'onChange', 
      control: {
        type: 'text'
      },
    },  
    children: {
      description:'children', 
      control: {
        type: 'text'
      },
    },
  }}
/>

### Checkbox

Use this playground to test the Checkbox component

<Canvas>
  <Story name="Default">
    {args => {
      return <Checkbox {...args} />;
    }}
  </Story>
</Canvas>

<ArgsTable story="Default" />