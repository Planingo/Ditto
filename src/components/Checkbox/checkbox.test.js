import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import {Checkbox} from './Checkbox';

describe('Checkbox component', () => {
  const onChange = jest.fn();
  const textContent = 'Coucou'
  const checkbox = (
    <Checkbox onChange={onChange}>{textContent}</Checkbox>
  );

  it('should render given children', () => {
    render(checkbox);
    expect(screen.getByText(textContent)).toBeInTheDocument();
  });

  it('should render given children', () => {
    render(checkbox);
    expect(screen.getByLabelText('checkbox')).toBeInTheDocument();
  });

  it('clicks on checkbox', () => {
    render(checkbox);
    const checkboxClick = screen.getByLabelText('checkbox');
    fireEvent.click(checkboxClick);
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});