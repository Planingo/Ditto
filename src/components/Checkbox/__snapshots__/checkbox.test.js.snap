// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Snapshot Checkbox 1`] = `
<body>
  <div>
    <label
      class="ant-checkbox-wrapper css-dev-only-do-not-override-j0nf2s"
    >
      <span
        class="ant-checkbox css-dev-only-do-not-override-j0nf2s"
      >
        <input
          aria-label="checkbox"
          class="ant-checkbox-input"
          type="checkbox"
          value=""
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        Coucou
      </span>
    </label>
  </div>
</body>
`;
